testdata/project-v4-multigroup/cmd/main.go

```diff
@@ -307,3 +326,39 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// checkAPIKindExists checks if a specific kind is available on the cluster across all groups and versions.
+func checkAPIKindExists(kind string) {
+	cfg, err := rest.InClusterConfig() // If running outside the cluster, use ctrl.GetConfigOrDie()
+	if err != nil {
+		setupLog.Error(err, "Unable to get Kubernetes config")
+		os.Exit(1)
+	}
+
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		setupLog.Error(err, "Unable to create discovery client")
+		os.Exit(1)
+	}
+
+	// Discover API groups and resources available on the cluster
+	_, resources, err := discoveryClient.ServerGroupsAndResources()
+	if err != nil {
+		setupLog.Error(err, "Unable to discover API groups and resources")
+		os.Exit(1)
+	}
+
+	// Check if the kind exists across all groups and versions
+	for _, resourceList := range resources {
+		for _, resource := range resourceList.APIResources {
+			if resource.Kind == kind {
+				setupLog.Info("API kind is installed on the cluster", "kind", kind)
+				return
+			}
+		}
+	}
+
+	// Log an error if the kind is not found
+	setupLog.Error(nil, fmt.Sprintf("API kind is not available on the cluster (kind: %s)", kind))
+	os.Exit(1)
+}
```

Comment:
Hi @fischor 

You made a comment that I lost regards you are not so happy with this part.
IHMO we should add the validation.
However, it need to be improved we should check the GKV and not only the Kind 
I am not happy with how it is done yet
So, we might can either let this part for an follow up where we discuss only the new markers and the inject code.